### PR TITLE
[WIP] Messaging index update

### LINKED ISSUE #2166: The page 'Messaging' requires a review
The page [Messaging](https://docs.particular.net/nservicebus/messaging/) has never been reviewed.

 * [Page review instructions](https://github.com/Particular/docs.particular.net#reviewing-a-page)
 * [Location in GitHub](https://github.com/Particular/docs.particular.net/blob/master/nservicebus/messaging/index.md)

Update page content to provide an introduction to messaging.

### PR DESCRIPTION
**Pull request by boblangley**
Closes #2166 

/cc @sfarmar

### CODE REVIEW

**Comment by boblangley** [comment]
This is still an incomplete and rough draft. I sent you an invite @sfarmar for the Lucid chart I created.

File: nservicebus/messaging/index.md [hunk 1]
```diff
@@ -1,4 +1,48 @@
...
+
+NServiceBus provides a means and standards for implementing indirect communication of messages between processes in .NET.
+
+## Physical, Logical, and Native Messages
+
+Messsaging protocols often make a distinction between a physical message and the logical message.
+
+The native message is the protocol/transport specific message format required by the underlying messaging technology.
+
+The physical message contains the logical message as well message metadata.
```
Comment (boblangley) [inline nservicebus/messaging/index.md]: This is the Transport Message in NServiceBus core, but is "physical" message the right descriptor? Or is it a synonym for the Native message (e.g. BrokeredMessage in ASB)?

File: nservicebus/messaging/index.md [hunk 2]
```diff
@@ -1,4 +1,48 @@
...
+
+Messages are discrete units of data communicated to one or more recipients using a predefined contract. Messages are a common way for systems or applications to interact, indeed the most common communications protocols on the Internet use messaging at their heart. HTTP protocol describes both a request message and the response message. Even SMTP is a messaging based system.
+
+Communication via messaging occurs either directly, as in the case of HTTP, or indirectly in the case of SMTP. The difference being whether the producer and/or consumer of the message is blocked during transfer and/or consumption of the message.
+
+NServiceBus provides a means and standards for implementing indirect communication of messages between processes in .NET.
+
+## Physical, Logical, and Native Messages
+
+Messsaging protocols often make a distinction between a physical message and the logical message.
```
Comment (boblangley) [inline nservicebus/messaging/index.md]: I added Native message late, so this sentence needs to be dropped or massaged

File: nservicebus/messaging/index.md [hunk 3]
```diff
@@ -1,4 +1,48 @@
...
+
+![Physical and Logical relationships](https://www.lucidchart.com/publicSegments/view/bf045538-99f3-4e49-9f79-7315d80fde61/image.png)
+
+## Defining Messages
+
+Messages in NServiceBus are defined by implementing a .NET class or interface with properties.  The namespace and type provide a way to distinguish one message from another.
+
+## Routing and Message Patterns
+
+Messages must know how to
```
Comment (boblangley) [inline nservicebus/messaging/index.md]: Ug, incomplete thought. Basically explain that routing allows the underlying transport to deliver the physical message to the correct queue.

File: nservicebus/messaging/index.md [hunk 4]
```diff
@@ -1,4 +1,55 @@
...
+The logical message is the portion of the physical message that contains the bytes that will actually be consumed by the user-code portion of the process, these bytes in NServiceBus being a serialized version of an object. 
+- Serialized / Deserialized
+
+![Physical and Logical relationships](https://www.lucidchart.com/publicSegments/view/bf045538-99f3-4e49-9f79-7315d80fde61/image.png)
+
+## Defining Messages
+
+Messages in NServiceBus are defined by implementing a .NET class or interface with properties.  The namespace and type provide a way to distinguish one message from another.
+
+## Routing and Message Patterns
```
Comment (weralabaj) [inline nservicebus/messaging/index.md]: Seems it's covering similar topics to https://docs.particular.net/nservicebus/architecture/principles#messaging-versus-rpc. Do we need to duplicate? Maybe somehow move out parts of the content from Principles page?

Comment (weralabaj) [inline nservicebus/messaging/index.md]: BTW I think this page shouldn't be where we explain concepts, it'll get very looooong and we already do it on the dedicated pages. I think we could simply provide a kind of a high-level map to guide the exploration.

Similar to what is done here https://docs.particular.net/get-started.

Thoughts?

Comment (SimonCropp) [inline nservicebus/messaging/index.md]: i have to agree with @weralabaj here

eg

> The logical message is the portion of the physical message that contains the bytes that will actually be consumed by the user-code portion of the process, these bytes in NServiceBus being a serialized version of an object.

this is way too low level for a index landing page

**Comment by boblangley** [comment]
I agree with the feedback so far. I am withdrawing the PR for now. I would like to submit another after we discuss requirements in #2166.

**Pull request closed without merging**